lsp-methods.el

```diff
@@ -1113,16 +1113,44 @@ https://github.com/Microsoft/language-server-protocol/blob/master/protocol.md#co
 	item))
     item))
 
+(defun lsp--extract-line-ref (buffer pos-start pos-end)
+  "Extract the reference's line and highlight it.
+
+BUFFER is the buffer visiting the file on which is the reference.
+If there is no visiting buffer, we don't extract the line and
+the function returns nil.
+
+POS-START and POS-END are LSP positions of the reference."
+  (when buffer
+    (with-current-buffer buffer
+      (let* ((start (lsp--position-to-point pos-start))
+             (end (lsp--position-to-point pos-end))
+             (line-range (save-excursion (goto-char start)
+                                         (cons (line-beginning-position)
+                                               (line-end-position))))
+             (line (buffer-substring (car line-range) (cdr line-range)))
+             (len (length line))
+             (start-on-line (- start (car line-range)))
+             (end-on-line (+ start-on-line (- end start))))
+        (add-face-text-property (max (min start-on-line len) 0)
+                                (max (min end-on-line len) 0)
+                                '(:box (:line-width -1 :color "white")) t line)
```

Comment:
Nit: maybe use `lsp-face-highlight-textual` here so that the lines match the in-buffer symbol highlighting.

Reply:
I don't think it's a good idea. I use a discret background for `lsp-face-highlight-textual`, not something really visible. In the contrary the reference should be very visible on the xref list.
Instead of box I use the face `highlight` now

Reply:
Ah, fair, I’ve got a pretty bold symbol highlight face.

Maybe a defcustom / new lsp face? This is the kinda thing I can see users wanting to tweak.

Reply:
Yes why not, but the face `highlight` is already customizable. Do you think it worth it to create a new face ?